apisix/http/router/radixtree_host_uri.lua

```diff
@@ -162,12 +161,12 @@ end
 function _M.matching(api_ctx)
     core.log.info("route match mode: radixtree_host_uri")
 
-    core.table.clear(match_opts)
+    local match_opts = core.tablepool.fetch("route_match_opts", 0, 16)
     match_opts.method = api_ctx.var.request_method
     match_opts.remote_addr = api_ctx.var.remote_addr
     match_opts.vars = api_ctx.var
     match_opts.host = api_ctx.var.host
-    match_opts.matched = core.tablepool.fetch("matched_route_record", 0, 4)
```

Comment:
Why do you modified this?

Reply:
Nesting memory requests from tablepool is a bad style and requires attention to the relationship between child and parent tables.
It may sacrifice some performance.

Reply:
I have decided to prioritize performance. 
udpate!